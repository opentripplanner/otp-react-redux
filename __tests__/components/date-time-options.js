import '../test-utils/mock-window-url'
import {
  getMockInitialState,
  mockWithProvider
} from '../test-utils/mock-data/store'
import { setDefaultTestTime } from '../test-utils'
import DateTimeOptions from '../../lib/components/form/call-taker/date-time-picker'

describe('components > form > call-taker > date time options', () => {
  beforeEach(setDefaultTestTime)

  // TODO: generate each of these with a method?
  it('should render', () => {
    const mockState = getMockInitialState()

    expect(
      mockWithProvider(
        DateTimeOptions,
        { date: '2022-11-17', time: '12:34' },
        mockState
      ).snapshot()
    ).toMatchSnapshot()
  })
  it('should correctly handle "12p"', () => {
    const mockState = getMockInitialState()

    expect(
      mockWithProvider(
        DateTimeOptions,
        { date: '2022-11-17', time: '12p' },
        mockState
      ).snapshot()
    ).toMatchSnapshot()
  })
  it('should correctly handle "12a"', () => {
    const mockState = getMockInitialState()

    expect(
      mockWithProvider(
        DateTimeOptions,
        { date: '2022-11-17', time: '12a' },
        mockState
      ).snapshot()
    ).toMatchSnapshot()
  })
  it('should correctly handle "1335"', () => {
    const mockState = getMockInitialState()

    expect(
      mockWithProvider(
        DateTimeOptions,
        { date: '2022-11-17', time: '1335' },
        mockState
      ).snapshot()
    ).toMatchSnapshot()
  })
  it('should correctly handle "135p"', () => {
    const mockState = getMockInitialState()

    expect(
      mockWithProvider(
        DateTimeOptions,
        { date: '2022-11-17', time: '135p' },
        mockState
      ).snapshot()
    ).toMatchSnapshot()
  })
  it('should correctly handle "133"', () => {
    const mockState = getMockInitialState()

    expect(
      mockWithProvider(
        DateTimeOptions,
        { date: '2022-11-17', time: '133' },
        mockState
      ).snapshot()
    ).toMatchSnapshot()
  })
  it('should correctly handle "133p"', () => {
    const mockState = getMockInitialState()

    expect(
      mockWithProvider(
        DateTimeOptions,
        { date: '2022-11-17', time: '133p' },
        mockState
      ).snapshot()
    ).toMatchSnapshot()
  })
})
