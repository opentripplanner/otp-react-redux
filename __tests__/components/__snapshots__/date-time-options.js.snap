// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > form > call-taker > date time options should correctly handle "12a" 1`] = `
<div>
  <select>
    <option
      value="NOW"
    >
      components.DateTimeOptions.now
    </option>
    <option
      value="DEPART"
    >
      components.DateTimeOptions.departAt
    </option>
    <option
      value="ARRIVE"
    >
      components.DateTimeOptions.arriveBy
    </option>
  </select>
  <input
    class="datetime-slim"
    style="line-height: .8em; margin-left: 3px; padding: 0px; width: 65px;"
    value="12:00 AM"
  />
  <input
    class="datetime-slim"
    style="font-size: 14px; line-height: 1em; outline: none; width: 120px;"
    type="date"
    value="2022-11-17"
  />
</div>
`;

exports[`components > form > call-taker > date time options should correctly handle "12p" 1`] = `
<div>
  <select>
    <option
      value="NOW"
    >
      components.DateTimeOptions.now
    </option>
    <option
      value="DEPART"
    >
      components.DateTimeOptions.departAt
    </option>
    <option
      value="ARRIVE"
    >
      components.DateTimeOptions.arriveBy
    </option>
  </select>
  <input
    class="datetime-slim"
    style="line-height: .8em; margin-left: 3px; padding: 0px; width: 65px;"
    value="12:00 PM"
  />
  <input
    class="datetime-slim"
    style="font-size: 14px; line-height: 1em; outline: none; width: 120px;"
    type="date"
    value="2022-11-17"
  />
</div>
`;

exports[`components > form > call-taker > date time options should correctly handle "133" 1`] = `
<div>
  <select>
    <option
      value="NOW"
    >
      components.DateTimeOptions.now
    </option>
    <option
      value="DEPART"
    >
      components.DateTimeOptions.departAt
    </option>
    <option
      value="ARRIVE"
    >
      components.DateTimeOptions.arriveBy
    </option>
  </select>
  <input
    class="datetime-slim"
    style="line-height: .8em; margin-left: 3px; padding: 0px; width: 65px;"
    value="1:03 PM"
  />
  <input
    class="datetime-slim"
    style="font-size: 14px; line-height: 1em; outline: none; width: 120px;"
    type="date"
    value="2022-11-17"
  />
</div>
`;

exports[`components > form > call-taker > date time options should correctly handle "133p" 1`] = `
<div>
  <select>
    <option
      value="NOW"
    >
      components.DateTimeOptions.now
    </option>
    <option
      value="DEPART"
    >
      components.DateTimeOptions.departAt
    </option>
    <option
      value="ARRIVE"
    >
      components.DateTimeOptions.arriveBy
    </option>
  </select>
  <input
    class="datetime-slim"
    style="line-height: .8em; margin-left: 3px; padding: 0px; width: 65px;"
    value="1:33 PM"
  />
  <input
    class="datetime-slim"
    style="font-size: 14px; line-height: 1em; outline: none; width: 120px;"
    type="date"
    value="2022-11-17"
  />
</div>
`;

exports[`components > form > call-taker > date time options should correctly handle "135p" 1`] = `
<div>
  <select>
    <option
      value="NOW"
    >
      components.DateTimeOptions.now
    </option>
    <option
      value="DEPART"
    >
      components.DateTimeOptions.departAt
    </option>
    <option
      value="ARRIVE"
    >
      components.DateTimeOptions.arriveBy
    </option>
  </select>
  <input
    class="datetime-slim"
    style="line-height: .8em; margin-left: 3px; padding: 0px; width: 65px;"
    value="1:35 PM"
  />
  <input
    class="datetime-slim"
    style="font-size: 14px; line-height: 1em; outline: none; width: 120px;"
    type="date"
    value="2022-11-17"
  />
</div>
`;

exports[`components > form > call-taker > date time options should correctly handle "1335" 1`] = `
<div>
  <select>
    <option
      value="NOW"
    >
      components.DateTimeOptions.now
    </option>
    <option
      value="DEPART"
    >
      components.DateTimeOptions.departAt
    </option>
    <option
      value="ARRIVE"
    >
      components.DateTimeOptions.arriveBy
    </option>
  </select>
  <input
    class="datetime-slim"
    style="line-height: .8em; margin-left: 3px; padding: 0px; width: 65px;"
    value="1:35 PM"
  />
  <input
    class="datetime-slim"
    style="font-size: 14px; line-height: 1em; outline: none; width: 120px;"
    type="date"
    value="2022-11-17"
  />
</div>
`;

exports[`components > form > call-taker > date time options should render 1`] = `
<div>
  <select>
    <option
      value="NOW"
    >
      components.DateTimeOptions.now
    </option>
    <option
      value="DEPART"
    >
      components.DateTimeOptions.departAt
    </option>
    <option
      value="ARRIVE"
    >
      components.DateTimeOptions.arriveBy
    </option>
  </select>
  <input
    class="datetime-slim"
    style="line-height: .8em; margin-left: 3px; padding: 0px; width: 65px;"
    value="12:34 PM"
  />
  <input
    class="datetime-slim"
    style="font-size: 14px; line-height: 1em; outline: none; width: 120px;"
    type="date"
    value="2022-11-17"
  />
</div>
`;
